torch/onnx/_internal/fx/passes/shape_inference.py

```diff
@@ -18,7 +18,12 @@ def shape_inference_with_fake_tensor(decomposed_module: torch.fx.GraphModule, *a
     # 2. run FakeTensorProp
     # If (1) and (2) are done with difference FakeTensorMode's, undefined behavior may
     # happen.
-    fake_tensor_mode = fake_tensor.FakeTensorMode()
+    fake_tensor_mode = torch._dynamo.utils.fake_mode_from_tensors(args)  # type: ignore[arg-type]
```

Comment:
nit
```suggestion
    fake_tensor_mode = torch._dynamo.utils.fake_mode_from_tensors(args) or fake_tensor.FakeTensorMode()
```

Reply:
I remember we want to use the `init_env` in FakeTensorMode to override some shapes. If we can't create FakeTensorMode here, then we need to modify FakeTensorMode's API to add a flag. @titaiwangms, for heads up.